libcudacxx/include/cuda/std/__linalg/conjugated.h

```diff
@@ -0,0 +1,158 @@
+//
+// THIS SOFTWARE IS PROVIDED BY SANDIA CORPORATION "AS IS" AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL SANDIA CORPORATION OR THE
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+// LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+// NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Questions? Contact Christian R. Trott (crtrott@sandia.gov)
+//
+// ************************************************************************
+//@HEADER
+*/
+#ifndef _LIBCUDACXX___LINALG_CONJUGATED_HPP
+#define _LIBCUDACXX___LINALG_CONJUGATED_HPP
+
+#include <cuda/std/detail/__config>
+
+#if defined(_CCCL_IMPLICIT_SYSTEM_HEADER_GCC)
+#  pragma GCC system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_CLANG)
+#  pragma clang system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_MSVC)
+#  pragma system_header
+#endif // no system header
+
+#include <cuda/std/version>
+
+#if defined(__cccl_lib_mdspan) && _CCCL_STD_VER >= 2017
+
+#  include <cuda/std/__linalg/conj_if_needed.h>
+#  include <cuda/std/__type_traits/add_const.h>
+#  include <cuda/std/__type_traits/is_arithmetic.h>
+#  include <cuda/std/__type_traits/remove_const.h>
+#  include <cuda/std/__utility/declval.h>
+#  include <cuda/std/mdspan>
+
+_LIBCUDACXX_BEGIN_NAMESPACE_STD
+
+namespace linalg
+{
+
+template <class _NestedAccessor>
+class conjugated_accessor
+{
+private:
+  using __nested_element_type = typename _NestedAccessor::element_type;
+  using __nc_result_type      = decltype(conj_if_needed(_CUDA_VSTD::declval<__nested_element_type>()));
+
+public:
+  using element_type     = _CUDA_VSTD::add_const_t<__nc_result_type>;
+  using reference        = _CUDA_VSTD::remove_const_t<element_type>;
+  using data_handle_type = typename _NestedAccessor::data_handle_type;
+  using offset_policy    = conjugated_accessor<typename _NestedAccessor::offset_policy>;
+
+  _CCCL_HIDE_FROM_ABI constexpr conjugated_accessor() = default;
+
+  _LIBCUDACXX_HIDE_FROM_ABI constexpr conjugated_accessor(const _NestedAccessor& __acc)
+      : __nested_accessor_(__acc)
+  {}
+
+  _CCCL_TEMPLATE(class _OtherNestedAccessor)
+  _CCCL_REQUIRES(_CCCL_TRAIT(_CUDA_VSTD::is_constructible, _NestedAccessor, const _OtherNestedAccessor&))
+  __MDSPAN_CONDITIONAL_EXPLICIT(!_CCCL_TRAIT(_CUDA_VSTD::is_convertible, _OtherNestedAccessor, _NestedAccessor))
+  _LIBCUDACXX_HIDE_FROM_ABI constexpr conjugated_accessor(const conjugated_accessor<_OtherNestedAccessor>& __other)
+      : __nested_accessor_(__other.nested_accessor())
+  {}
```

Comment:
I do not believe that works properly because that just drops the explicit on the floor if there is no compiler support. 

We should rather bite the bullet and duplicate the constructor once explicit and once not